shared/src/commonMain/kotlin/datasources/HabitLocalDataSource.kt

```diff
@@ -25,10 +23,10 @@ class HabitLocalDataSource(database: HabitDatabase): HabitDataSource {
 
     override suspend fun addHabit(habit: HabitRowData) = withContext(Dispatchers.Unconfined) {
         habitQueries.insertHabit(habit.name)
-        val habitId = habitQueries.getHabitId(habit.name).executeAsOne() //unwraps the Query<> part
+        val habitId = habitQueries.getHabitId(habit.name).executeAsOne() //unwraps the Query<> part?
         val lastFiveDates = getLastFiveDates()
         val dateToStatusMap = lastFiveDates.zip(habit.lastFiveDayStatuses).toMap()
-        val datesToInsert = dateToStatusMap.filter { it.value == true }.keys
+        val datesToInsert = dateToStatusMap.filter { it.value == true }.keys //== for explicit clarity that "it" is a map with Boolean values
```

Comment:
If you feel it helpful, you can add an alias to Lamda functions like this.

`dataToStatusMap.filter{ status -> status.value == true }`

Reply:
I will have to look up aliases, this i didn't like the Boolean expression alone because `it ` looks strange to me still, when i look at just `it.value` (so i added the comment haha) I will update it though.